dashboard/src/modules/components/TableOfContent/index.jsx

```diff
@@ -0,0 +1,414 @@
+        <div className="toc">
+          <br />
+
+          <Menu
+            id="rootMenu"
+            containsDrilldown
+            drilldownItemPath={drilldownPath}
+            drilledInMenus={menuDrilledIn}
+            activeMenu={activeMenu}
+            onDrillIn={drillIn}
+            onDrillOut={drillOut}
+            onGetMenuHeight={setHeight}
+          >
+            {breadCrumb && (
+              <>
+                <MenuBreadcrumb>{breadCrumb}</MenuBreadcrumb>
+                <Divider component="li" />
+              </>
+            )}
+            {isLoading ? (
+              <Spinner className="spinner"></Spinner>
+            ) : (
+              <MenuContent>
+                <MenuList>
+                  {contentData?.directories?.map((data, index) => {
+                    return (
+                      <MenuItem
+                        itemId="group:start_rollout"
+                        id="d_down_parent"
+                        key={index}
+                        direction="down"
+                        onClick={() => {
+                          breadCrumbLabels.push(data);
+                          setBreadCrumbLabels(breadCrumbLabels);
+                          setBreadCrumb(initialBreadcrumb(breadCrumbLabels));
+                          const dirPath = param.concat(data);
+                          setParam(dirPath);
+                          setIsLoading(true);
+                          getSubFolderData(dirPath);
+                        }}
+                        drilldownMenu={
+                          <DrilldownMenu id="drilldownMenuStart">
+                            {currData?.directories?.map((data, index) => {
+                              if (dirCount < currData.directories.length) {
+                                dirCount = dirCount + 1;
+                                return (
+                                  <MenuItem
+                                    itemId="dir_info"
+                                    id="d_down"
+                                    key={index}
+                                    direction="down"
+                                    onClick={() => {
+                                      breadCrumbLabels.push(data);
+                                      setBreadCrumbLabels(breadCrumbLabels);
+                                      setBreadCrumb(
+                                        appGroupingBreadcrumb(
+                                          false,
+                                          breadCrumbLabels
+                                        )
+                                      );
+                                      const dirPath = param.concat("/", data);
+                                      setParam(dirPath);
+                                      setIsLoading(true);
+                                      getSubFolderData(dirPath);
+                                    }}
```

Comment:
This code looks remarkably similar to the code at lines 249-257, seems like an opportunity for refactoring.